ci_framework/roles/reproducer/tasks/libvirt_layout.yml

```diff
@@ -0,0 +1,213 @@
+---
+- name: Deploy layout on target host
+  tags:
+    - libvirt_layout
+  ansible.builtin.import_role:
+    name: libvirt_manager
+    tasks_from: deploy_layout
+
+- name: Push generated inventory from hypervisor
+  ansible.builtin.command:  # noqa: command-instead-of-module
+    cmd: >-
+      rsync -r {{ cifmw_reproducer_basedir }}/reproducer-inventory/
+      zuul@controller-0:reproducer-inventory
+
+- name: Slurp ssh key for CRC access
+  when:
+    - _layout.vms.crc is defined
+  tags:
+    - bootstrap
+    - bootstrap_layout
+  register: crc_priv_key
+  ansible.builtin.slurp:
+    path: "{{ ansible_user_dir}}/.crc/machines/crc/id_ecdsa"
+
+- name: Configure CRC services
+  when:
+    - _layout.vms.crc is defined
+  tags:
+    - bootstrap
+    - bootstrap_layout
+  delegate_to: crc-0
+  remote_user: root
+  block:
+    - name: Ensure crc-0 knows about its second NIC
+      when:
+        - cifmw_job_uri is undefined
+      community.general.nmcli:
+        autoconnect: true
+        conn_name: private_net
+        dns4: 127.0.0.1
+        ifname: enp2s0
+        type: ethernet
+        ip4: "{{ cifmw_reproducer_crc_ip4 }}/24"
+        gw4: "{{ cifmw_reproducer_crc_gw4 }}"
+        state: present
+
+    - name: Ensure crc-0 does not get "public" DNS
+      community.general.nmcli:
+        autoconnect: true
+        conn_name: "Wired connection 1"
+        dns4_ignore_auto: true
+        state: present
+
+    - name: Configure dns forwarders
+      ansible.builtin.blockinfile:
+        path: "/var/srv/dnsmasq.conf"
+        block: |-
+          {% if cifmw_reproducer_dns_servers %}
+          {% for dns_server in cifmw_reproducer_dns_servers %}server={{ dns_server }}{% endfor %}
+          {% endif %}
+
+    - name: Configure local DNS for CRC pod
+      register: last_modification
+      ansible.builtin.replace:
+        path: "/var/srv/dnsmasq.conf"
+        regexp: "192.168.130.11"
```

Comment:
This one could be a variable. In the case of changes happening to `install_yamls` (I'm not sure how often it changes though) it would make easier adjustments.

Reply:
for now it's hard-coded all the way down. There are further tasks in order to decouple things from install_yamls.